File: backend/src/main/java/com/yzh/yingshi/entity/Device.java
package com.yzh.yingshi.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("device")
public class Device {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String deviceSerial;

    private Integer channelNo;

    private String deviceName;

    private String deviceType;

    private String sourceType;

    private String streamUrl;

    private String status;

    private String remark;

    @TableLogic
    private Integer deleted;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}

File: frontend/src/types/device.ts
export interface DeviceVO {
  id: number;
  deviceSerial: string;
  channelNo: number;
  deviceName: string;
  deviceType: string;
  sourceType: string;
  status: 'ONLINE' | 'OFFLINE' | 'DISABLED';
  remark: string;
  createdAt: string;
  updatedAt: string;
}

export interface DeviceUpdateDTO {
  deviceName: string;
  remark?: string;
  status?: string;
}

export interface DeviceSyncResultDTO {
  total: number;
  inserted: number;
  updated: number;
  message: string;
}
